cli/command/trust/signer_remove.go

```diff
@@ -0,0 +1,124 @@
+package trust
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	"github.com/spf13/cobra"
+)
+
+type signerRemoveOptions struct {
+	forceYes bool
+	images   []string
+	signer   string
+}
+
+func newSignerRemoveCommand(dockerCli command.Cli) *cobra.Command {
+	options := signerRemoveOptions{}
+	cmd := &cobra.Command{
+		Use:   "signer-remove NAME IMAGE [IMAGE...]",
+		Short: "Remove a signer",
+		Args:  cli.RequiresMinArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.signer = args[0]
+			options.images = args[1:]
+			return removeSigner(dockerCli, &options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.forceYes, "yes", "y", false, "Answer yes to removing most recent signer (no confirmation)")
+	return cmd
+}
+
+func removeSigner(cli command.Cli, options *signerRemoveOptions) error {
+	for _, image := range options.images {
+		if err := removeSingleSigner(cli, image, options.signer, options.forceYes); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isLastSigner(role data.Role, allRoles []client.RoleWithSignatures) (bool, error) {
+	var lastSigner client.RoleWithSignatures
+	for _, role := range allRoles {
+		if role.Name == releasesRoleTUFName {
+			lastSigner = role
+		}
+	}
+	if len(lastSigner.Signatures) == 0 {
+		return false, fmt.Errorf("All signed tags are currently revoked, use docker trust sign to fix")
+	}
+	for _, signature := range lastSigner.Signatures {
+		for _, key := range role.KeyIDs {
+			if signature.KeyID != key {
```

Comment:
I think this should instead check `if signature.KeyID == key` and technically also check against the threshold?  We should count the number of keys from the role that have signed the  threshold and `return false, error` if `len(Signatures) - count < threshold`

Reply:
I think ```signature.KeyID != key``` is correct. Wouldn't this require knowing ucp signing policy?

Reply:
from IRL discussion: as @eiais mentions, we should increment on `signature.KeyID != key`.  This is TUF thresholding separate from UCP signing policy